Task5.java

```diff
@@ -0,0 +1,114 @@
+
+        products = new String[]{"apple", "orange", "lemon", "banana", "apricot", "avocado",
+                "broccoli", "carrot", "cherry", "garlic", "grape", "melon", "leek", "kiwi",
+                "mango", "mushroom", "nut", "olive", "pea", "peanut", "pear", "pepper",
+                "pineapple", "pumpkin", "potato"};
+
+        poundArray = new String[]{"#","#","#","#","#","#","#","#","#","#","#","#","#","#","#"};
+
+        computerProductInput();
+        userProductInput();
+    }
+
+    //Random fixed choice of a product by the computer
+    public static void computerProductInput(){
+        Random random = new Random();
+        randomIndex = random.nextInt(products.length);
+        computerProduct = products[randomIndex];
+    }
+
+    //Initial input of a product by the user
+    public static void userProductInput(){
+        sc = new Scanner(System.in);
+        System.out.println("Hi! Let's play a guessing game!\n" +
+                "I have a list of " + products.length +" products:" );
+        displayProducts();
+        System.out.println("\nI chose one product from the list.\n" +
+                           "(Computer chose '" + computerProduct + "', but don't tell it to the user:))\n"+
+                           "Try to guess which one. Enter here:");
+        userProduct = sc.nextLine();
+    }
+
+    //Symbol by symbol matching of the computer product with the user product.
+    //If the matching fails, the user has attempts to input another product
+    //as many times as needed to guess the computer product.
+    public static void matchComputerAndUserProducts(){
+        while (true){
+            try {
+                for (int i = 0; i < computerProduct.length(); i++){
+                    if (computerProduct.charAt(i) == userProduct.charAt(i)){
+                        poundArray[i] = String.valueOf(computerProduct.charAt(i));
+                    }
+                }
+            } catch (StringIndexOutOfBoundsException ignored) {}
+
+            if(computerProduct.equals(userProduct)){
+                System.out.println("Congratulations! You won! " + "'" + computerProduct + "'"
+                        + " is the right product:)");
+                sc.close();
+                break;
+            } else {
+                userAnotherTry();
+            }
+        }
+    }
+
+    //Display of the product array
+    public static void displayProducts(){
+        for (String s : products) {
+            System.out.print(s + ", ");
+        }
+    }
+    //Display of the pound(#) array
+    public static void displayPoundArray (){
+        for (String s : poundArray) {
+            System.out.print(s);
+        }
+    }
+
+    //Another input of a product by the user,
+    //reinitialization of the pound(#) array.
+    public static void userAnotherTry(){
+        displayPoundArray();
+        poundArray = new String[]{"#","#","#","#","#","#","#","#","#","#","#","#","#","#","#"};

```

Comment:
why do you initialize poundArray after each try? Is it possible to simplify it somehow?